Source/WebCore/dom/Element.cpp

```diff
@@ -2068,13 +2067,17 @@ ExceptionOr<void> Element::setAttribute(const AtomString& qualifiedName, const T
     if (!document().scriptExecutionContext()->settingsValues().trustedTypesEnabled)
         setAttributeInternal(index, name, std::get<AtomString>(value), InSynchronizationOfLazyAttribute::No);
     else {
-        auto attributeTypeAndSink = trustedTypeForAttribute(nodeName(), name.localName().convertToASCIILowercase(), this->namespaceURI(), name.namespaceURI());
+        AttributeTypeAndSink attributeTypeAndSink;
+        if (document().requiresTrustedTypes())
+            attributeTypeAndSink = trustedTypeForAttribute(nodeName(), name.localName().convertToASCIILowercase(), this->namespaceURI(), name.namespaceURI());
         auto attributeValue = trustedTypesCompliantAttributeValue(attributeTypeAndSink.attributeType, value, this, attributeTypeAndSink.sink);
```

Comment:
Do we need to invoke this when trusted types are not used? I would just change `if (!document().scriptExecutionContext()->settingsValues().trustedTypesEnabled)` to `if (document().requiresTrustedTypes())` above if not.

Reply:
Yes because we need to extract the string from the TrustedType objects passed through to the function.

Even if TT isn't enforced in CSP they can still create the objects so we need to support that scenario.